src/ticks.jl

```diff
@@ -203,19 +204,22 @@ function optimize_ticks_typed(
     sigdigits(z) = max(1, x_digits - z + q_extra_digits)
 
     ib = Int(base)
-    round_base = (
-        isinteger(base) ? v -> round(v, sigdigits = sigdigits(z), base = ib) :
-        v -> round(v, sigdigits = sigdigits(z))
-    )
+    round_base = let z_ = z
+        (
+            isinteger(base) ? v -> round(v, sigdigits = sigdigits(z_), base = ib) :
+            v -> round(v, sigdigits = sigdigits(z_))
+        )
+    end
```

Comment:
```suggestion
    # generalizing "order of magnitude"
    z::Int = bounding_order_of_magnitude(xspan, base)

    # find required significant digits for ticks with q * base^z spacing,
    # for q values specified in Q
    x_digits::Int = bounding_order_of_magnitude(max(abs(x_min), abs(x_max)), base)
    q_extra_digits::Int = maximum(postdecimal_digits(q[1]) for q in Q)

    sigdigits::Int = max(1, x_digits - z + q_extra_digits)

    round_base = v -> round(
        v,
        sigdigits = sigdigits,
        base = isinteger(base) ? Int(base) : base
    )
    end
```

Adding type assertion helps resolving type instability here.

Reply:
Those variables were inferred fine for me, I don't have any red (except non-performance critical for the `@warn` macro) in `@code_warntype` anymore. The main problem was the forced boxing of `z` through its inclusing in closures.

Reply:
That looks reasonable, thanks for the link. I'm sure we can rewrite that in a cleaner way (if CI is willing to cooperate !).